File: solutions_5658068650033152_0/java/Mistyepd/Enclosure.java

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Scanner;
/**
 * @author Kate Henderson
 *
 */
public class Enclosure {

	private Scanner sc = null;
	private BufferedWriter bw = null;	
	private File inFile;
	private File outFile;
	private int cases;

	private boolean[][] game;
	private int rows;
	private int cols;
	private int K;
	private int min;



	public Enclosure(){
		inFile = new File("C-small-attempt0.in");
		outFile = new File("C-small-attempt0.out");
	}

	public static void main(String[] args) {
		Enclosure p = new Enclosure(); 
		p.solve();	
	}

	private void solve(){		
		try {
			sc = new Scanner(new FileReader(inFile));
			bw = new BufferedWriter(new FileWriter(outFile));

			cases = sc.nextInt();

			for (int c = 1; c <= cases; c++) {
				StringBuilder sb = new StringBuilder();
				rows = sc.nextInt();
				cols = sc.nextInt();
				K = sc.nextInt();
				game = new boolean[cols][rows];
				int space = rows*cols;
				/*for (int y = 0; y<rows; y++){
					for (int x = 0; x<cols; x++){
						game[x][y];
					}
				}*/
				if((space-4)==K)
					sb.append("Case #" + c + ": "+((rows*2)+(cols*2)-8)+"\n");
				else if (rows<=2||cols<=2){
					sb.append("Case #" + c + ": "+K+"\n");
				}else{
					int small = Math.min(rows, cols);
					if (K%2==0){
						min = K/2+2;
					}else{
						int cur = (int)K-((K+1)/3 *2);
						min = ((K+1)/3 *2)+cur;
					}				
					sb.append("Case #" + c + ": "+min+"\n");
				}
				try {
					bw.write(sb.toString());
				} catch (IOException e) {
					e.printStackTrace();
				}
			}

			bw.flush();
			bw.close();
		} catch (IOException e) {
			e.printStackTrace();
		}
	}

}

File: solutions_5658068650033152_0/Python/Zexin/Enclosure.py
import math

infile = open('C-small-attempt1.in', 'r')
outfile = open('try.out', 'w')

st = infile.readline()
st = st.rstrip()
n = int(st)

for m in range(1, n+1):
    st = infile.readline()
    st = st.rstrip()
    sp = st.split(' ')
    outfile.write('Case #' + str(m) + ': ')
    a = int(sp[0])
    b = int(sp[1])
    k = int(sp[2])
    if a > b:
        s = a
        a = b
        b = s
    mint = 99999
    for i in range(k, a*b+1):
        x = i + 4
        y = x
        for j in range(int(math.sqrt(x)), 0, -1):
            if x % j == 0:
                x = x // j - 2
                if j > 2:
                    y = j - 2
                else:
                    y = 0
                break
        if x > y:
            s = x
            x = y
            y = s
        if (y < b) and (x < a):
            t = (x + y) * 2
        else:
            t = 999999
        # print('i : ', i, '  x : ', x, '  y : ', y, '  t :', t)
        if mint > t:
            mint = t
    if mint == 99999:
        if a > 2:
            a1 = a - 2
        else:
            a1 = 0
        if b > 2:
            b1 = b - 2
        else:
            b1 = 0
        mint = a*b - a1*b1
    outfile.write(str(mint) + '\n')

infile.close()
outfile.close()
